### ISSUE TITLE
As a user, I want to be able to view a LD resource in a nice web page, so I can participate in LD goodness

### ISSUE THREAD

**Issue by jblom** [beeldengeluid/beng-lod-server#48]
**Approach**
- make sure you can run the LOD server on your localhost
- edit the server.py and add an extra route to the new HTML page
```
@app.route('/test-rdfa')
def resource():
	return render_template('resource.html')
```
- add the file `resource.html` to src/templates (you have to create the `templates` dir)
- add a new css file to `src/static/css` (you have to create the `static/css` dir in `src`)
- if you have the LOD server running, you can access your resource.html page via `http://localhost:xxxx/test-rdfa`
 
Now you can start working on the HTML page that should:
- read data from a single resource from the LOD server (as JSON) via `http://localhost:xxxx/resource/<level>/<identifier>` (use accept type= `application/ld+json`)
- transforms this data into [RDFa](https://www.w3.org/TR/xhtml-rdfa-primer/) elements on the page
- then create a CSS that nicely styles the RDFa elements on the screen, you can import the css via `/static/css/yourfile.css` 

**Acceptance criteria**
- [ ] follow the steps above to create a nice HTML page that can display the data of a single resource
- [ ] don't use any javascript frameworks, but go for the pure HTML/CSS/Javascript approach

**Comment by avlisodraude** [comment]
@mwigham @jblom 
Issue for beng-lod-server project done in branch:
_issue/#48-add-html-view-mode-to-resource_

**GitHub:**
_https://github.com/beeldengeluid/beng-lod-server/tree/issue/%2348-add-html-view-mode-to-resource_

### What's been done?
A template to render resources based on level and id has been added along with the needed functionality to make it retrieve the data requested.

### How to test?
- Update Master branch since it has been updated to make use of python 3.
- Switch to the working branch (issue/#48-add-html-view-mode-to-resource).
- From the src folder run `pipenv run python server.py`
- Once your localhost is running point to it assuming 5309 is used as the port:
    http://0.0.0.0:5309/resource-html/program/5289392
- The first parameter (program) correspond to the 'level' and the second is the identifier.

**Comment by mwigham** [comment]
Tried to test this but unfortunately I get the following error:

```requests.exceptions.ConnectionError: HTTPConnectionPool(host='0.0.0.0', port=5309): Max retries exceeded with url: /resource/program/5289392 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x00000201316A7AC8>: Failed to establish a new connection: [WinError 10049] Het aangevraagde adres is niet geldig in de context van het adres'))```

Do you have an idea what could be causing this please?

**Comment by jblom** [comment]
@mwigham it's probably the 0.0.0.0:5309 thing. Just paste the URL path `/resource-html/program/5289392` after the host+port of your own local lod server.

**Comment by mwigham** [comment]
@jblom no, that's not it unfortunately. localhost:5310 (that's the port on my laptop) shows me the API with no problems, but when I paste the URL path after it, I get the error

**Comment by mwigham** [comment]
@avlisodraude do you know what the problem could be, perhaps?

**Comment by mwigham** [comment]
@avlisodraude  I got this working. I don't know to what extent the issues I faced will affect others, but here is a list:

server.py
- The requestUrl was hardcoded using 0.0.0.0, my setup won't accept that. It also hardcoded the port, leading to discrepancies between the port it was running on, and the port it was trying to access
- The requestUrl had '{level}' and '{identifier}' hardcoded so that these were passed instead of the actual values
- the keys were out of date with the current version of the schema, they used 'schema' for the namespace instead of 'nisv',  'carrierclass' should be 'hasCarrierClass',  and 'creationdate' 'hasCreationDate'

resource.html
- same issue with the concept names being out-of-date.

For the rest, I tested with a couple of programmes, and a season, series and segment. It worked nicely! It makes it much easier to understand the item and see the relation between the parts of it. I only have two minor comments:
- perhaps use hasSortDate instead of hasCreationDate - the former is when it was broadcast, the latter is for the carrier only
- creator needs to be changed in the template to take into account that this now refers to a skos concept, so the name needs to be looked up from the skos concept
